build-script-helper.py

```diff
@@ -127,12 +135,21 @@ def run(args):
       verbose=args.verbose)
 
 
-def build_package(package_dir, swift_build_exec, swiftc_exec, sourcekit_searchpath, swiftsyntax_searchpath, build_dir, config='release', verbose=False):
+def build_package(package_dir, swift_build_exec, swiftc_exec, parser_header_dir, parser_lib_dir, sourcekit_searchpath, swiftsyntax_searchpath, build_dir, config='release', verbose=False):
   env = dict(os.environ)
   env['SWIFT_EXEC'] = swiftc_exec
 
   swiftc_args = ['-lSwiftSyntax', '-I', swiftsyntax_searchpath, '-L', swiftsyntax_searchpath, '-Fsystem', sourcekit_searchpath]
   linker_args = ['-rpath', swiftsyntax_searchpath, '-rpath', sourcekit_searchpath]
+
+  # To find the syntax parser library.
+  if parser_header_dir:
+    swiftc_args.extend(['-I', parser_header_dir])
+  if parser_lib_dir:
+    swiftc_args.extend(['-L', parser_lib_dir])
+    if platform.system() == 'Darwin':
+      linker_args.extend(['-rpath', parser_lib_dir])
```

Comment:
I don't think it's necessary to check the platform here—we don't currently support building either tool outside Darwin, and the rest of build-script-helper assumes we need to worry about rpaths. On the other hand, it doesn't really hurt to have it.